Result of `nixpkgs-review pr 214720` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.ruff-lsp</li>
    <li>python310Packages.ruff-lsp.dist</li>
    <li>python311Packages.ruff-lsp</li>
    <li>python311Packages.ruff-lsp.dist</li>
  </ul>
</details>

Tests are failing:
```
       > +                  {'code': 'F821',
       > +                   'data': {'fix': None, 'noqa_row': 3},
       > +                   'message': 'Undefined name `x`',
       > +                   'range': {'end': {'character': 7, 'line': 2},
       > +                             'start': {'character': 6, 'line': 2}},
       > +                   'severity': 1,
       > +                   'source': 'Ruff'}],
       > - {'diagnostics': [], 'uri': 'file:///build/tmp7_ngkmwz.py'}
       > ? -------------------
       >
       > +  'uri': 'file:///build/tmp7_ngkmwz.py'}
       >
       > ----------------------------------------------------------------------
       > Ran 2 tests in 0.828s
       >
       > FAILED (failures=2)
       > /nix/store/5s1yg5l36wzgy1dj0vv1ibarc4g7vrdr-stdenv-linux/setup: line 1594: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/sf96dvqm6k82hz8qvfpq8cdhvamsr76s-python3.10-ruff-lsp-0.0.24.drv'.
error: builder for '/nix/store/2y40r03xspznxfqr8pd968i6i3mvn1zf-python3.11-ruff-lsp-0.0.24.drv' failed with exit code 1;
+                   'range': {'end': {'character': 10, 'line': 0},
+                             'start': {'character': 7, 'line': 0}},
+                   'severity': 2,
+                   'source': 'Ruff',
+                   'tags': [1]},
+                  {'code': 'F821',
+                   'data': {'fix': None, 'noqa_row': 3},
+                   'message': 'Undefined name `x`',
+                   'range': {'end': {'character': 7, 'line': 2},
+                             'start': {'character': 6, 'line': 2}},
+                   'severity': 1,
+                   'source': 'Ruff'}],
- {'diagnostics': [], 'uri': 'file:///build/tmp7mmxyo8r.py'}
? -------------------

+  'uri': 'file:///build/tmp7mmxyo8r.py'}

======================================================================
FAIL: test_no_initialization_options (tests.test_server.TestServer.test_no_initialization_options)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/ruff_lsp-0.0.24/tests/test_server.py", line 181, in test_no_initialization_options
    self.assertEqual(actual, expected)
AssertionError: {'uri[46 chars]s': []} != {'uri[46 chars]s': [{'range': {'start': {'line': 0, 'characte[533 chars]f'}]}
+ {'diagnostics': [{'code': 'F401',
+                   'data': {'fix': {'edits': [{'content': '',
+                                               'end_location': {'column': 0,
+                                                                'row': 2},
+                                               'location': {'column': 0,
+                                                            'row': 1}}],
+                                    'message': 'Remove unused import: `sys`'},
+                            'noqa_row': 1},
+                   'message': '`sys` imported but unused',
+                   'range': {'end': {'character': 10, 'line': 0},
+                             'start': {'character': 7, 'line': 0}},
+                   'severity': 2,
+                   'source': 'Ruff',
+                   'tags': [1]},
+                  {'code': 'F821',
+                   'data': {'fix': None, 'noqa_row': 3},
+                   'message': 'Undefined name `x`',
+                   'range': {'end': {'character': 7, 'line': 2},
+                             'start': {'character': 6, 'line': 2}},
+                   'severity': 1,
+                   'source': 'Ruff'}],
- {'diagnostics': [], 'uri': 'file:///build/tmppxpuymvg.py'}
? -------------------

+  'uri': 'file:///build/tmppxpuymvg.py'}

----------------------------------------------------------------------
Ran 2 tests in 0.722s

FAILED (failures=2)
```

--- a/pkgs/development/python-modules/ruff-lsp/default.nix
+++ b/pkgs/development/python-modules/ruff-lsp/default.nix
@@ -1,0 +1,63 @@
+{ lib
+, stdenv
+, pythonOlder
+, buildPythonPackage
+, fetchPypi
+, ruff
+, pygls
+, lsprotocol
+, hatchling
+, typing-extensions
+, unittestCheckHook
+, python-lsp-jsonrpc
+}:
+
+buildPythonPackage rec {
+  pname = "ruff-lsp";
+  version = "0.0.24";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "ruff_lsp";
+    sha256 = "sha256-1he/GYk8O9LqPXH3mu7eGWuRygiDG1OnJ+JNT2Pynzo=";
+  };
+
+  postPatch = ''
+    # ruff binary added to PATH in wrapper so it's not needed
+    sed -i '/"ruff>=/d' pyproject.toml
+  '';
+
+  nativeBuildInputs = [
+    hatchling
+  ];
+
+  propagatedBuildInputs = [
+    pygls
+    lsprotocol
+    typing-extensions
+  ];
+
+  doCheck = stdenv.isDarwin;
+
+  nativeCheckInputs = [
+    unittestCheckHook
+    python-lsp-jsonrpc
+    ruff
+  ];
+
+  makeWrapperArgs = [
+    # prefer ruff from user's PATH, that's usually desired behavior
+    "--suffix PATH : ${lib.makeBinPath [ ruff ]}"
+  ];
+
+
+  meta = with lib; {
+    homepage = "https://github.com/charliermarsh/ruff-lsp";
+    description = "A Language Server Protocol implementation for Ruff";
+    changelog = "https://github.com/charliermarsh/ruff-lsp/releases/tag/v${version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ kalekseev ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -10457,6 +10457,8 @@ self: super: with self; {
 
   rubymarshal = callPackage ../development/python-modules/rubymarshal { };
 
+  ruff-lsp = callPackage ../development/python-modules/ruff-lsp { };
+
   ruffus = callPackage ../development/python-modules/ruffus { };
 
   runway-python = callPackage ../development/python-modules/runway-python { };

